zypp/KeyRing.cc

```diff
@@ -512,22 +543,30 @@ namespace zypp
     if ( foundKey ) {
       // it exists, is trusted, does it validate?
       context_r.signatureIdTrusted( whichKeyring == trustedKeyRing() );
-      report->infoVerify( filedesc, foundKey, context );
+      report->infoVerify( filedesc, foundKey, keyContext );
       if ( verifyFile( file, signature, whichKeyring ) )
       {
 	context_r.fileValidated( true );
+	if ( context_r.signatureIdTrusted() && not buddies.empty() ) {
+	  // Check for buddy keys to be imported...
+	  MIL << "Validated with trusted key: importing buddy list..." << endl;
+	  for ( auto kd : buddies ) {
```

Comment:
shouldn't that be const auto & ?  Otherwise you are copying all of them.

Reply:
done